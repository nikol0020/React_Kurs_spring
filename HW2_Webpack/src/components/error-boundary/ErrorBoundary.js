import React, {Component} from 'react';
import './ErrorBoundary.scss';

export default class ErrorBoundary extends Component {
    constructor(props) {
        super(props);
        this.state = { hasError: false };
    }

    static getDerivedStateFromError(error) {
        return { hasError: true };
    }

    componentDidCatch(error, info) {
        console.log('information about error', error, info);
    }

    render() {
        if (this.state.hasError) {
            return <h1>Something went wrong.</h1>;
        }
        return this.props.children;
    }
};