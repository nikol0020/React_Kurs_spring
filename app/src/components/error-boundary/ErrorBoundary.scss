h1 {
  margin: 0 auto;
  color: #f00;
}