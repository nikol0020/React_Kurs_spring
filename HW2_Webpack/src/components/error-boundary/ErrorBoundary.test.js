import React from 'react';
import ErrorBoundary from './ErrorBoundary';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });

describe('<ErrorBoundary /> shallow rendering', () => {

    it('should render ErrorBoundary', () => {
        const wrapper = shallow(<ErrorBoundary />);
        expect(wrapper).toMatchSnapshot();
    });

});