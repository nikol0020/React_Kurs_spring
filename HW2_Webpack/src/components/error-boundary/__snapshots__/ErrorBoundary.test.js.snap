// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> shallow rendering should render ErrorBoundary 1`] = `""`;
